micrometer-java11/src/main/java/io/micrometer/java11/instrument/binder/jdk/MicrometerHttpClient.java

```diff
@@ -277,6 +270,9 @@ public <T> CompletableFuture<HttpResponse<T>> sendAsync(HttpRequest httpRequest,
             }
             instrumentation.setResponse(response);
             stopObservationOrTimer(instrumentation, request, response);
+            if (throwable != null) {
+                throw new CompletionException(throwable);
```

Comment:
This is the important change for the original issue.

Reply:
Is this the proper way to do this or does `CompletableFuture` let us to signal an error on it instead?
(Though, since this is a "handle" method, I would assume it's ok to throw an exception from it.)

Reply:
I'm not sure there's another way to do it. This results in the same outcome as without the handler, I believe. Exceptions thrown will always be wrapped in `CompletionException` as far as I know.